org-brain.el

```diff
@@ -62,24 +64,94 @@ This will be used by `org-brain-new-child'."
   :group 'org-brain
   :type '(string))
 
+;;; Utils
+(defun flatten (obj)
+  (do* ((result (list obj))
+        (node result))
+       ((null node) (delete nil result))
+    (cond ((consp (car node))
+           (when (cdar node) (push (cdar node) (cdr node)))
+           (setf (car node) (caar node)))
+          (t (setf node (cdr node))))))
+
+;;; Logging
+(defcustom org-brain-log t
+  "Set to nil to not write to *Messages* buffer."
+  :group 'org-brain
+  :type 'boolean)
+
+(defun org-brain-log (msg)
+  (when org-brain-log
+    (setq inhibit-message t)
+    (message msg)
+    (setq inhibit-message nil)))
+
+;;; Caches
+(setq org-brain-files-cache nil)
+(setq org-brain-parents-cache nil)
+(setq org-brain-children-cache nil)
+(setq org-brain-pins-cache nil)
+
+(defun org-brain-invalidate-all-caches ()
+  "This is a convenience function for those who (occasionally)
+  edit children, parents, i.e., entries in the Brainchildren
+  node, or pins, outside the org-brain-visualize interface. In
+  that case, you have to call this function manually. It is not
+  needed if children, parents, and pins are added using the
+  org-brain-visualize interface/mode."
+  (interactive)
+  (setq org-brain-files-cache nil)
+  (setq org-brain-parents-cache nil)
+  (setq org-brain-children-cache nil)
+  (setq org-brain-pins-cache nil))
+
+(defun org-brain-invalidate-files-cache ()
```

Comment:
Again not quite necessary (since the names of the functions are clear), but could have docstrings.